### PR TITLE
[Nettle] add cflags to build tools

### PR DESCRIPTION
**Pull request by xandox**
Add cflags to build command line for build tools, for PR #15605 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  same, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 Yes

### CODE REVIEW

File: ports/nettle/flags.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/Makefile.in b/Makefile.in
+index e5ccfc7..922aa0c 100644
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -309,7 +309,7 @@ $(LIBHOGWEED_FORLINK): $(hogweed_OBJS) $(LIBNETTLE_FORLINK)
+ 
+ # For building the various *data.c programs. -lm needed for shadata.
+ %$(EXEEXT_FOR_BUILD): %.c
+-	$(CC_FOR_BUILD) $< -lm -o $@
++	$(CC_FOR_BUILD) $< $(CFLAGS) -lm -o $@
```
Comment (JackBoosY) [inline ports/nettle/flags.patch:10]: Can I know what's the problem you met?

Comment (xandox) [reply ports/nettle/flags.patch:10]: Same thing, package compiles tools for build without any CFLAGS (or CFLAGS_FOR_BUILD). It's ok for windows and linux, but for osx not.

Comment (xandox) [reply ports/nettle/flags.patch:10]: [build-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/6357166/build-x64-osx-dbg-out.log)

Comment (xandox) [reply ports/nettle/flags.patch:10]: [build-x64-osx-dbg-err.log](https://github.com/microsoft/vcpkg/files/6357169/build-x64-osx-dbg-err.log)

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
The osx regression is not related to this issue, waiting for rerun.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**